GVRf/Framework/src/org/gearvrf/GVRViewManager.java

```diff
@@ -883,4 +885,24 @@ public GVREventManager getEventManager() {
     public GVRScriptManager getScriptManager() {
         return mScriptManager;
     }
+
+
+    protected long mGlDeleterPtr;
+
+    @Override
+    public void finalize() throws Throwable {
+        try {
+            if (0 != mGlDeleterPtr) {
```

Comment:
It looks like the DeletePtr requires explicit life-cycle management in user code, and is scatter in different places (a raw pointer, a constructor invocation, a destructor invocation, and using the raw pointer in API). This seems to be something for encapsulation.